src/Components/Facility/DischargeModal.tsx

```diff
@@ -169,6 +174,15 @@ const DischargeModal = ({
 
   const prescriptionActions = PrescriptionActions(consultationData.id ?? "");
 
+  const handleFacilitySelect = (selected: FacilityModel) => {
+    setFacility(selected ? selected : facility);
+    const { name } = selected;
+    setPreDischargeForm((prev) => ({
+      ...prev,
+      referred_to_external: name || "",
```

Comment:
The purpose of `referred_to_external` is to store facilities which are not present in our system. In case the facility does exist, we are to store it in `referred_to` and leave `referred_to_external` blank. You are storing the selected facility unconditionally as `referred_to_external` without checking if it is indeed external facility or not. 

Look at the code implementaion here for reference on how to do the above check: 
https://github.com/coronasafe/care_fe/blob/develop/src/Components/Facility/ConsultationForm.tsx#L744-L758

In summary:
- User selects a facility which is in our system -> send it as `referred_to: "facility-id-here"`
- User types and selects a facility which is not in our system -> send it as `referred_to_external: "name of facility"`